Doofinder/Feed/Controller/Adminhtml/Integration/CreateStore.php

```diff
@@ -4,292 +4,86 @@
-        $this->attributeCollectionFactory = $attributeCollectionFactory;
-        $this->cacheFrontendPool = $cacheFrontendPool;
         parent::__construct($context);
     }
 
     /**
      * @inheritDoc
-     *
-     * @throws WebapiException
      */
     public function execute()
     {
         $resultJson = $this->resultJsonFactory->create();
 
-        if ($this->generateDoofinderStores() == true) {
-            $resultJson->setData(true);
-        } else {
-            $resultJson->setHttpResponseCode(WebapiException::HTTP_INTERNAL_ERROR);
-            $resultJson->setData(false);
-        }
-        return $resultJson;
-    }
+        try {
+            $installationResults = $this->installationService->generateDoofinderStores();
 
-    public function generateDoofinderStores()
-    {
-        $success = true;
-        foreach ($this->storeConfig->getAllGroups() as $storeGroup) {
-            try {
-                $storeGroupId = (int)$storeGroup->getId();
-                $websiteId = (int)$storeGroup->getWebsiteId();
-                $searchEngineData = $this->generateSearchEngineData($storeGroupId);
-                $storeOptions = $this->generateStoreOptions($websiteId);
-                $primary_language = $this->storeConfig->getLanguageFromStore($storeGroup->getDefaultStore());
+            $installationFailed = true;
 
-                $storeGroupConfig = [
-                    "name" => $storeGroup->getName(),
-                    "platform" => "magento2",
-                    "primary_language" => $primary_language,
-                    "skip_indexation" => false,
-                    "sector" => $this->storeConfig->getValueFromConfig(StoreConfig::SECTOR_VALUE_CONFIG),
-                    "site_url" => $this->get_primary_site_url_in_se($searchEngineData["searchEngineConfig"], $primary_language),
-                    "search_engines" => $searchEngineData["searchEngineConfig"],
-                    "options" => $storeOptions,
-                    "query_input" => "#search",
-                    "plugin_version" => $this->getModuleVersion()
-                ];
-                $response = $this->storeConfig->createStore($storeGroupConfig);
-                $this->saveInstallationConfig((int)$storeGroupId, $response["installation_id"], $response["script"]);
-                $this->saveSearchEngineConfig($searchEngineData["storesConfig"], $response["config"]["search_engines"]);
-            } catch (Exception $e) {
-                $success = false;
-                $this->logger->error('Error creating store for store group "' . $storeGroup->getName() .
-                    '". ' . $e->getMessage());
+            $message = 'Doofinder stores generated successfully.';
+            foreach ($installationResults as $result) {
+                if ($result !== true) {
+                    $message = 'The installation was not being fully successful. Please check the logs for further information.';
```

Comment:
I might be wrong, but for me this sentence sounds more natural I think:

```suggestion
                    $message = __('The installation was not fully successful. Please check the logs for further information.');
```